hosted/aks/p1/p1_provisioning_test.go

```diff
@@ -728,4 +748,66 @@ var _ = Describe("P1Provisioning", func() {
 		}
 	})
 
+	XContext("Private Cluster", func() {
+		// Blocked on: https://github.com/rancher/rancher/issues/43772
+		BeforeEach(func() {
+			var err error
+			k8sVersion, err = helper.GetK8sVersion(ctx.RancherAdminClient, ctx.CloudCredID, location, true)
+			Expect(err).NotTo(HaveOccurred())
+			GinkgoLogr.Info(fmt.Sprintf("Using K8s version %s for cluster %s", k8sVersion, clusterName))
+
+			createFunc := func(clusterConfig *aks.ClusterConfig) {
+				clusterConfig.PrivateCluster = pointer.Bool(true)
+			}
+
+			cluster, err = helper.CreateAKSHostedCluster(ctx.RancherAdminClient, clusterName, ctx.CloudCredID, k8sVersion, location, createFunc)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				cluster, err = ctx.RancherAdminClient.Management.Cluster.ByID(cluster.ID)
+				Expect(err).To(BeNil())
+				return cluster.Transitioning == "error" && strings.Contains(cluster.TransitioningMessage, "failed to communicate with cluster: error generating service account token") && strings.Contains(cluster.TransitioningMessage, "cluster agent disconnected")
```

Comment:
Both error strings appear at the same time? shouldn't be OR used here instead?

Reply:
>Both error strings appear at the same time?

They are part of the same error.
"tcp handshake timeout" error occurs at times instead of `cluster agent disconnected`.